qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,224 @@
+
+
+def run(test, params, env):
+    """
+    Block performance test with fio
+    Steps:
+    1) boot up guest with one data disk on specified backend and pin qemu-kvm process to the last numa node on host
+    2) pin guest vcpu and vhost threads to cpus of last numa node repectively
+    3) format data disk and run fio in guest
+    4) collect fio results and host info
+    """
+    def cpu_thread(cpu_cmd, cpufile):
+        cpu_cmd = "%s > %s" % (cpu_cmd, cpufile)
+        commands.getoutput(cpu_cmd)
+
+    def fio_thread(session, run_cmd, cmd_timeout):
+        session.cmd_status(run_cmd, cmd_timeout)
+
+    def _pin_vm_threads(vm, node):
+        if node:
+            if not isinstance(node, utils_misc.NumaNode):
+                node = utils_misc.NumaNode(int(node))
+            utils_test.qemu.pin_vm_threads(vm, node)
+
+        return node
+
+    # login virtual machine
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+    logging.debug(commands.getoutput("numactl --hardware"))
+    logging.debug(commands.getoutput("numactl --show"))
+    _pin_vm_threads(vm, params.get("numa_node"))
+
+    # get parameter from dictionary
+    fio_cmd = params.get('fio_cmd')
+    write_cmd = params.get('write_cmd')
+    rw = params.get("rw")
+    block_size = params.get("block_size")
+    iodepth = params.get("iodepth")
+    threads = params.get("threads")
+    cmd_timeout = int(params.get("cmd_timeout", 1200))
+    order_list = params.get("order_list")
+    driver_format = params.get("drive_format")
+    guest_ver_cmd = params.get("guest_ver_cmd")
+    pattern = params.get("pattern")
+    pre_cmd = params.get("pre_cmd")
+    format = params.get("format")
+    md5value = params.get("md5value")
+    compile_cmd = params.get("compile_cmd")
+    test_src = params.get("test_src")
+    os_type = params.get("os_type")
+    image_format = params.get("image_format")
+    drop_cache = params.get("drop_cache")
+
+    result_path = utils_misc.get_path(test.resultsdir,
+                                      "fio_result.RHS")
+    result_file = open(result_path, "w")
+
+    # scratch host and windows guest version info
+    get_version(session, result_file, guest_ver_cmd, os_type,
+                driver_format, cmd_timeout)
+
+    # online disk
+    if os_type == "windows":
+        for num in range(1, 2):
+            disks = check_status(session, cmd_timeout, num)
+            diskstatus = re.findall("Disk\s+\d+\s+(\w+).*?\s+\d+", disks[0])[0]
+            if diskstatus == "Offline":
+                online_disk_cmd = params.get("online_disk_cmd")
+                online_disk_run = online_disk_cmd % num
+                (s, o) = session.cmd_status_output(online_disk_run,
+                                                   timeout=cmd_timeout)
+                if s != 0:
```

Comment:
CODING_STYLE: Use "is None" rather than "== None" and "is not None" rather than "!= None".